### ISSUE TITLE
HTTP Compression not working!

### ISSUE THREAD

**Issue by osrec** [swoole/swoole-src#2974]
I tried the below script using Swoole 4.4.7, 4.4.8 and also the latest code in the master branch. I can't seem to get compression to work.

My test code is as follows:

```php
$server = new Swoole\Http\Server("0.0.0.0", 8888, SWOOLE_BASE);

$server->set(
[
	'backlog'	=> 1000,
	'max_conn' => 1024,
	'max_request'	=> 1000,
	'timeout' => 30,
	'worker_num' => 4,
	'enable_reuse_port' => true,
	'open_tcp_keepalive' => true,
	'daemonise' => true,
	'display_errors' => true,
	'log_level' => 1,
	'task_worker_num' => 0,
	'http_compression' => true,
	'http_gzip_level' => 9,

]);

$server->on('request', function ($request, $response) use ($server)
{
	print_r($request);
	$data = [];
	for ($i = 0; $i < 1000; ++$i) {	$data[$i] = md5($i);  }
	$response->end(json_encode($data));
	print_r($response);
});

$server->start();
```

I see no compression in the responses - only uncompressed data. Could it be that I need to compile swoole with a particular dependency that I'm missing?

Thanks!

**Comment by twose** [comment]
show your `php --ri swoole` please

respect our issue template

**Comment by osrec** [comment]
Hi @twose, please see the below output of `php --ri swoole`

```
swoole

Swoole => enabled
Author => Swoole Team <team@swoole.com>
Version => 4.4.7
Built => Oct  9 2019 02:23:29
coroutine => enabled
epoll => enabled
eventfd => enabled
signalfd => enabled
cpu_affinity => enabled
spinlock => enabled
rwlock => enabled
sockets => enabled
openssl => OpenSSL 1.1.1c  28 May 2019
http2 => enabled
pcre => enabled
mutex_timedlock => enabled
pthread_barrier => enabled
futex => enabled
mysqlnd => enabled
async_redis => enabled

Directive => Local Value => Master Value
swoole.enable_coroutine => On => On
swoole.enable_library => On => On
swoole.enable_preemptive_scheduler => Off => Off
swoole.display_errors => On => On
swoole.use_shortname => On => On
swoole.unixsock_buffer_size => 8388608 => 8388608
```

**Comment by twose** [comment]
zlib is missing

please install zlib by apt/yum/brew first

**Comment by osrec** [comment]
@twose - thank you - this now appears to be resolved in the code in the master branch - for anyone else having this issue, I had to install the `zlib1g-dev` package (`sudo apt-get install zlib1g-dev`) in order to get this to work.

**Comment by twose** [comment]
The fix about zlib in the master branch is not related to your problem

Just install `zlib1g-dev` and it can work well all of the version

**Comment by osrec** [comment]
> The fix about zlib in the master branch is not related to your problem
> 
> Just install `zlib1g-dev` and it can work well all of the version

Thanks. Yes, I believe the fix relates to POST requests, which is what my code was actually using (my example code above was a bit simplified compared to my actual use case).

Thanks for all your help :)

**Comment by gggeek** [comment]
Sorry to revive an old thread, but I just spent one day trying to figure out why the CI tests for my app using Swoole were failing on GitHub Actions and passing locally (the test env being built in both cases 100% from custom code and not relying on 3rd party GH Actions).

I think the difference is that, at the time Swoole is built (using PIE) on the GH runner, the `zlib1g-dev` is present, but it is not when building in my local containers.

Which then makes Swoole not able to use zlib to send back compressed responses, even if the zlib extension is enabled in php.

What made this hard to troubleshoot is that, when Swoole has no support for zlib, running `php --ri swoole` does _not_ show anything as being missing / disabled.

Long story short: would it be possible to have swoole output `zlib => disabled` instead of not showing anything, in case it was compiled without it?

(of course this applies as well to any other swoole not-enabled feature)
